### PR TITLE
restore setEnvWithDotEnv

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
restore `setEnvWithDotEnv` to load .env into os.Env so the many places we rely on os.LookupEnv aren't borken
this also restore support for configuring buildx using env var set in .env

includes e2e test + a tiny fix to include (stopped) one-off containers as "orphans"

**Related issue**
closes https://github.com/docker/compose/issues/11967

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**
image

### CODE REVIEW

File: pkg/compose/containers.go [hunk 1]
```diff
@@ -118,17 +118,16 @@ func isRunning() containerPredicate {
...
-		return !utils.StringContains(services, service)
-	}
-}
-
 // isOrphaned is a predicate to select containers without a matching service definition in compose project
 func isOrphaned(project *types.Project) containerPredicate {
 	services := append(project.ServiceNames(), project.DisabledServiceNames()...)
 	return func(c moby.Container) bool {
+		// One-off container
+		v, ok := c.Labels[api.OneoffLabel]
```
Comment (jhrotko) [inline pkg/compose/containers.go:126]: question: what is the oneOffLabel?

Comment (ndeloof) [reply pkg/compose/containers.go:126]: this is used to distinguish containers created by  `docker compose run` vs actual services created by `up`

File: pkg/e2e/compose_run_test.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func TestLocalComposeRun(t *testing.T) {
 			"Hello one more time")
 		lines = Lines(res.Stdout())
 		assert.Equal(t, lines[len(lines)-1], "Hello one more time", res.Stdout())
-		assert.Assert(t, !strings.Contains(res.Combined(), "orphan"))
+		assert.Assert(t, strings.Contains(res.Combined(), "orphan"))
```
Comment (jhrotko) [inline pkg/e2e/compose_run_test.go:40]: question: was this a bug? :o

**Review by jhrotko — APPROVED** [review]
LGTM

File: pkg/e2e/compose_run_test.go (hunk 2 above)
Comment (ndeloof) [reply pkg/e2e/compose_run_test.go:40]: more or less. We only considered "orphans" for service container not declared in current compose model. This PR also consider stopped `run` containers as orphaned

**Review by glours — APPROVED** [review]

**Pull request merged**